sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -41,10 +41,12 @@ trait StringRegexExpression extends Expression
 
   override def dataType: DataType = BooleanType
 
-  // try cache the pattern for Literal
+  // try cache foldable pattern
   private lazy val cache: Pattern = pattern match {
-    case Literal(value: String, StringType) => compile(value)
```

Comment:
@MaxGekk . This is SPARK-9242 (Spark 1.5.0) https://github.com/apache/spark/pull/8321/files#diff-39298b470865a4cbc67398a4ea11e767R41.

Reply:
Reynold just moved the code. Actually it exists since 1.0.0: https://github.com/apache/spark/commit/af3746ce0d724dc624658a2187bde188ab26d084#diff-d788f93e29b4d25cdd7d60328587678bR42